src/datasets/streaming.py

```diff
@@ -85,20 +84,29 @@ def wrapper(*args, **kwargs):
     patch_submodule(module, "os.path.isdir", wrap_auth(xisdir)).start()
     patch_submodule(module, "os.path.isfile", wrap_auth(xisfile)).start()
     patch_submodule(module, "os.path.getsize", wrap_auth(xgetsize)).start()
-    if hasattr(module, "Path"):
-        patch.object(module.Path, "joinpath", xpathjoin).start()
-        patch.object(module.Path, "__truediv__", xpathjoin).start()
-        patch.object(module.Path, "open", wrap_auth(xpathopen)).start()
-        patch.object(module.Path, "glob", wrap_auth(xpathglob)).start()
-        patch.object(module.Path, "rglob", wrap_auth(xpathrglob)).start()
-        patch.object(module.Path, "parent", property(fget=xpathparent)).start()
-        patch.object(module.Path, "name", property(fget=xpathname)).start()
-        patch.object(module.Path, "stem", property(fget=xpathstem)).start()
-        patch.object(module.Path, "suffix", property(fget=xpathsuffix)).start()
-    patch_submodule(module, "pd.read_csv", wrap_auth(xpandas_read_csv), attrs=["__version__"]).start()
-    patch_submodule(module, "pd.read_excel", xpandas_read_excel, attrs=["__version__"]).start()
-    patch_submodule(module, "sio.loadmat", wrap_auth(xsio_loadmat), attrs=["__version__"]).start()
-    # xml.etree.ElementTree
-    for submodule in ["ElementTree", "ET"]:
-        patch_submodule(module, f"{submodule}.parse", wrap_auth(xet_parse)).start()
+    patch_submodule(module, "pathlib.Path", xPath).start()
```

Comment:
Now we can just pass the source object to patch in the module, and it will patch it even the attribute or a parent module has been renamed (see test_patching.py for a list of all supported cases - I probably have to add a docstring to `patch_submodule` as well)